src/GPG.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PharIo\GnuPG;
+
+interface GPG
+{
+    /**
+     * @param string $keyData raw key data
+     *
+     * @throws InvalidHomeDirectory
+     * @throws InvalidKey
+     */
+    public function importPublicKey(string $keyData): PublicKey;
+
+    /**
+     *
+     * @since 2.1 this method can be added later because phive is no just using the verify logic
+     * @param string $keyData raw key data
+     *
+     * @throws InvalidHomeDirectory
+     * @throws InvalidKey
+     */
+    public function importSecretKey(string $keyData): SecretKey;
+
+    /**
+     * Verifies the message is signed with the sigature
+     *
+     * Ensures that the signature is valid.
+     *
+     * @throws VerificiationFailed When message was not signed with the signature
+     * @throws UnknownFingerPrint When signature doesn't belong to a known key, key must be imported first
+     */
+    public function verify(string $message, Signature $signature): KeyInfo;
```

Comment:
This doesn't feel like the best API we can do yet:

- It's a common and (sadly?) expected scenario that the public key is unknown and thus the verification will fail. So this should not be an exception.

- Also failing the signature check is a valid scenario and thus does not merit an exception. 

- To avoid control flow via exceptions, only getting a boolean false from the pecl variant / an actual error exit code from cli should trigger exceptions.

- `KeyInfo` is the wrong return type for this method, imho. We get verification specific results from gnupg like the timestamp of the signature creation, keyid and fingerprint used and whether or not the signature check performed succeeded. We should reflect this in our API.